src/NServiceBus.AwsLambda.SQS.Analyzer/AwsLambdaDiagnostics.cs

```diff
@@ -0,0 +1,111 @@
+             title: "LimitMessageProcessing is not supported in Aws Lambda",
+             messageFormat: "Concurrency-related settings can be configured with the Lambda API.",
+             category: DiagnosticCategory,
+             defaultSeverity: DiagnosticSeverity.Error,
+             isEnabledByDefault: true
+            );
+
+        internal static readonly DiagnosticDescriptor DefineCriticalErrorActionNotAllowed = new DiagnosticDescriptor(
+             id: DefineCriticalErrorActionNotAllowedId,
+             title: "DefineCriticalErrorAction is not supported in Aws Lambda",
+             messageFormat: "Aws Lambda endpoints do not control the application lifecycle and should not define behavior in the case of critical errors.",
+             category: DiagnosticCategory,
+             defaultSeverity: DiagnosticSeverity.Error,
+             isEnabledByDefault: true
+            );
+
+        internal static readonly DiagnosticDescriptor SetDiagnosticsPathNotAllowed = new DiagnosticDescriptor(
+             id: SetDiagnosticsPathNotAllowedId,
+             title: "SetDiagnosticsPath is not supported in Aws Lambda",
+             messageFormat: "Aws Lambda endpoints should not write diagnostics to the local file system. Use CustomDiagnosticsWriter to write diagnostics to another location.",
+             category: DiagnosticCategory,
+             defaultSeverity: DiagnosticSeverity.Error,
+             isEnabledByDefault: true
+            );
+
+        internal static readonly DiagnosticDescriptor MakeInstanceUniquelyAddressableNotAllowed = new DiagnosticDescriptor(
+             id: MakeInstanceUniquelyAddressableNotAllowedId,
+             title: "MakeInstanceUniquelyAddressable is not supported in Aws Lambda",
+             messageFormat: "Aws Lambda endpoints have unpredictable lifecycles and should not be uniquely addressable.",
+             category: DiagnosticCategory,
+             defaultSeverity: DiagnosticSeverity.Error,
+             isEnabledByDefault: true
+            );
+
+        internal static readonly DiagnosticDescriptor UseTransportNotAllowed = new DiagnosticDescriptor(
+             id: UseTransportNotAllowedId,
+             title: "UseTransport is not supported in Aws Lambda",
+             messageFormat: "The package configures Amazon SQS transport by default. Use ServiceBusTriggeredEndpointConfiguration.Transport to access the transport configuration.",
+             category: DiagnosticCategory,
+             defaultSeverity: DiagnosticSeverity.Warning,
+             isEnabledByDefault: true
+            );
+
+        internal static readonly DiagnosticDescriptor OverrideLocalAddressNotAllowed = new DiagnosticDescriptor(
+             id: OverrideLocalAddressNotAllowedId,
+             title: "OverrideLocalAddress is not supported in Aws Lambda",
+             messageFormat: "The NServiceBus endpoint address in Aws Lambda is determined by the ServiceBusTrigger attribute.",
```

Comment:
ServiceBusTrigger is the Azure Functions way. Need to figure out the equivalent Lambda address binding mechanism.